### PR TITLE
expanded documentation on how to handle rejections

### PR DESCRIPTION
**Pull request by aujxn**
I saw an easy open issue #126 to expand the docs about handling rejections. I'm new to the library but would like to get involved with the project.

### CODE REVIEW

File: src/reject.rs [hunk 1]
```diff
@@ -11,20 +11,56 @@
...
+//!
+//! // Custom rejection handler that maps rejections into responses.
+//! async fn handle_rejection(err: Rejection) -> Result<impl Reply, std::convert::Infallible> {
+//!     if err.is_not_found() {
+//!         Ok(reply::with_status("NOT_FOUND", StatusCode::NOT_FOUND))
+//!     } else if let Some(e) = err.find::<InvalidParameter>() {
+//!         Ok(reply::with_status("BAD_REQUEST", StatusCode::BAD_REQUEST))
+//!     } else {
+//!         // Could use eprintln! if not using logging crate
+//!         error!("unhandled rejection: {:?}", err);
```
Comment (jxs) [inline src/reject.rs:42]: we can just `eprintln!` here, and remove the comment

File: src/reject.rs [hunk 2]
```diff
@@ -11,20 +11,56 @@
...
+//! async fn main() {
+//!
+//!     // Filter on `/:id`, but reject with InvalidParameter if the `id` is `0`.
+//!     // Recover from this rejection using a custom rejection handler.
+//!     let route = warp::path::param()
+//!         .and_then(|id: u32| async move {
+//!             if id == 0 {
+//!                 Err(warp::reject::custom(InvalidParameter))
+//!             } else {
+//!                 Ok("something since id is valid")
```
Comment (jxs) [inline src/reject.rs:57]: we can just have "id is valid!" here

**Review by jxs — CHANGES_REQUESTED** [review]
Hi, and thanks for getting this started! :+1:  Only missing fixing the issues reported by CI and a couple of notes I left

File: src/reject.rs (hunk 1 above)
Comment (aujxn) [reply src/reject.rs]: Yeah, this makes more sense thanks for the help

File: src/reject.rs (hunk 2 above)
Comment (aujxn) [reply src/reject.rs]: Perfect

**Comment by aujxn** [comment]
Awesome! Thanks for the help, `cargo test` succeeds now I should have tried that before submitting the PR.

**Review by jxs — APPROVED** [review]

**Comment by jxs** [comment]
awesome, thanks! :rocket:

**Pull request merged**